src/const_eval.md

```diff
@@ -61,40 +61,22 @@ A _const context_ is one of the following:
 * A [const generic argument]
 * A [const block]
 
+Const contexts that are used as parts of types (array type and repeat length
+expressions as well as const generic arguments) can only make restricted use of
+surrounding generic type and lifetime parameters.
```

Comment:
Do you happen to know what the restrictions are?

Also, just to verify, there are no restrictions for const generic parameters now?

Reply:
I don't know, I just copied this text to a place where I think it makes more sense.
Cc @BoxyUwU

Reply:
type system const expressions can either be arbitrary expressions not making using of any generics, or a single bare const parameter, e.g. `{ N }`

Reply:
Thanks! I've added this to the PR.

It's not related to "const fn float" at all, but I don't want to make this a separate PR as those would conflict...